bokeh/util/logconfig.py

```diff
@@ -18,9 +18,36 @@
 from __future__ import absolute_import
 
 import logging
+import sys
 
 from ..settings import settings
 
 level = settings.py_log_level()
+bokeh_logger = logging.getLogger('bokeh')
+root_logger = logging.getLogger()
+
 if level is not None:
-    logging.basicConfig(level=level)
+    bokeh_logger.setLevel(level)
+
+if not (root_logger.handlers or bokeh_logger.handlers):
+    # No handlers configured => at least add a printer to sys.stderr for
```

Comment:
Sorry, I'm still abit lost about parts of the situation. I guess a more direct question is: What can be put in `bokeh/__init__.py` unconditionally to make sure these messages never fall in a black hole  (even for "library" usage of Bokeh) without unduly affecting things.